src/main/java/com/zendesk/maxwell/zk/ZkService.java

```diff
@@ -0,0 +1,134 @@
+package com.zendesk.maxwell.zk;
+
+import org.I0Itec.zkclient.IZkStateListener;
+import org.I0Itec.zkclient.ZkClient;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.zookeeper.Watcher;
```

Comment:
Maxwell currently doesn't have any dependency on zookeeper. Given it does have a mysql dependency, why not use mysql locking as suggested by @osheroff rather than adding a dependency on zookeeper.